### ISSUE TITLE
a ** 2.0 has incorrect type (migrated from Trac #2033)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3580]
Original ticket http://projects.scipy.org/numpy/ticket/2033
Reported 2012-01-30 by trac user kasal, assigned to atmention:teoliphant.

Hello,
I observe an incorrect evaluation of the power operator for the exponent = 2.0:
    print 'NumPy:', np.**version**
    a = np.array([2, 4], dtype=np.int16)
    for e in np.arange(5.0):
        s = 'a**{}'.format(e)
        print '{:9} {}'.format(s, repr(eval(s)))

```
!NumPy: 1.6.0
a**0.0    array([ 1.,  1.])
a**1.0    array([ 2.,  4.])
a**2.0    array([ 4, 16], dtype=int16)
a**3.0    array([  8.,  64.])
a**4.0    array([  16.,  256.])
```

In all other cases, the output dtype is float, because of te float exponent.  But 'a*_2.0' seems to be incorrectly reduced to 'a_a'.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-01-30

Confirmed for 2.0.0.dev-473f125 as well.

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2012-02-11

This is the result of fast_scalar_power which special cases the 2.0 case to multiply by itself (which in this case keeps the dtype what it used to be) instead of upgrading it.  

This should be fixed. 

-Travis

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2012-02-13

Fixed in Pull Request #207

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2012-02-14
